ui/src/message_popup/linear_session/linear_session.jsx

```diff
@@ -57,9 +69,13 @@ export default React.createClass({
   renderContent() {
     const {questions} = this.props;
     const {responses} = this.state;
-    if (responses.length >= questions.length) return this.props.summaryEl(questions, responses);
 
-    const question = questions[responses.length];
+    if (this.props.getNextQuestion === undefined || this.props.getNextQuestion(questions, responses) === null) {
```

Comment:
With `getDeaultProps`, I don't think you need to do this check for whether it's undefined.   Also, the way this is written now you could call `getNextQuestion` twice: one here and once on line 77.

I think it'd be clearer using something like `const question = this.props.getNextQuestion(...)` and then checking that variable's value.